### ISSUE TITLE
Clippy does not check all features

### ISSUE THREAD

**Issue by EricLBuehler** [rust-lang/rust#117075]
I tried this code:

```rust
#![allow(dead_code)]

#![deny(clippy::all)]
#![warn(clippy::pedantic)]
#![warn(clippy::nursery)]
#![warn(clippy::cargo)]

#[cfg(bland_feature)]
struct AltStruct {
    data: Arc<i32>,
}

struct Struct {
    data: i32,
}

fn main() {

}
```

I expected an error saying that Arc is not found in scope, but there is no error.

Is this expected behavior? Is there an easier way to do cargo clippy on all features?

**Comment by matthiaskrgr** [comment]
Does it work with `cargo clippy --all-features`?

**Comment by EricLBuehler** [comment]
Unfortunately, no.

**Comment by fmease** [comment]
`bland_feature` is not a Cargo feature but a configuration option. To set the configuration when running Clippy, execute `RUSTFLAGS='--cfg bland_feature' cargo clippy`. There's no way for Clippy or rustc to know all possible configuration options, therefore there's no flag to enable "all of them" (there's [`check_cfg`] with which you specify all legal options manually but that's unstable).

The easiest option would be to turn the cfg option into a Cargo feature by adding `bland_feature = []` to section `[features]` in your `Cargo.toml` and updating the attribute to `#[cfg(feature = "bland_feature")]`. This way, you can utilize `cargo clippy --all-features` (and `cargo clippy --features bland_feature`).

**Comment by Nilstrieb** [comment]
also note that normal cfgs are often not additive, so enabling all of them will often lead to bad results.

**Comment by EricLBuehler** [comment]
> `bland_feature` is not a Cargo feature but a configuration option. To set the configuration when running Clippy, execute `RUSTFLAGS='--cfg bland_feature' cargo clippy`. There's no way for Clippy or rustc to know all possible configuration options, therefore there's no flag to enable "all of them" (there's [`check_cfg`](https://doc.rust-lang.org/nightly/unstable-book/compiler-flags/check-cfg.html) with which you specify all legal options manually but that's unstable). Edit: rustc could theoretically look through all `#[cfg{,_attr}]`s to determine a set of options but that's not how the language works.
> 
> The easiest option would be to turn the cfg option into a Cargo feature by adding `bland_feature = []` to section `[features]` in your `Cargo.toml` and updating the attribute to `#[cfg(feature = "bland_feature")]`. This way, you can utilize `cargo clippy --all-features` (and `cargo clippy --features bland_feature`).

Thank you! Changing `#[cfg(bland_feature)]` to `#[cfg(feature = "bland_feature")]` got the behavior I wanted.

How would I have enabled the original `cfg(bland_feature)`?

**Comment by fmease** [comment]
Closing as wontfix / discussion resolved.
